### PR TITLE
Cleanup common defines

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

It is a header that included in almost all compilation units, so changing it requires recompile everything.
So instead:
- move some macros into separate headers
- remove some macros (because they are not used)
- inline some macros into settings values (no need in extra hop for looking the value)
- move some macros into better place (macros for ASan [un]poisoning)

### CODE REVIEW

File: base/base/defines.h [hunk 1]
```diff
@@ -31,6 +31,8 @@
 #   endif
 #endif
 
+// more aliases: https://mailman.videolan.org/pipermail/x264-devel/2014-May/010660.html
+
 #if defined(_MSC_VER)
```
Comment (alexey-milovidov) [inline base/base/defines.h:36]: BTW, we don't even pretend to work on Windows, we can remove this macro.

**Review by alexey-milovidov — APPROVED** [review]
LGTM

File: base/base/defines.h (hunk 1 above)
Comment (azat) [reply base/base/defines.h:36]: Completely agree. Will fix it shortly.

**Comment by azat** [comment]
>Stress test (thread) — Sanitizer assert (in stderr.log)

#29692

**Pull request merged**